src/twister2/device/device_abstract.py

```diff
@@ -37,19 +36,10 @@ def connect(self, timeout: float = 1) -> None:
     def disconnect(self) -> None:
         pass
 
-    def flash(self, build_dir: str | Path, timeout: float = 60.0) -> None:
+    def flash_and_run(self, timeout: float = 60.0) -> None:
```

Comment:
I thought we want to have `flash` and `run` as separated methods because for simulator we only run the code.

Reply:
Yes, I know, that I requested for splitting flash and run into two separate methods. But I came into consideration, that it was a mistake. Probably there is no sense for splitting those two actions into two separate methods and enforce on potential users to remember every time, when they use `DeviceAbstract` object to call those two methods to be able to be compatible with running tests both on hardware and simulator. That's why I unified those two methods into one. But you are right that naming it `flash...` is misfortune because in simulations there is no flashing action. But I don't have better idea how to name it correctly and to be still consistent with what this method does.

Reply:
Ok, it is fine for me. Possibly we could have method name like `execute`, but it is less clear what is going on under the hood.